src/coins.cpp

```diff
@@ -1158,6 +1158,12 @@ CValidationState::Code CCoinsViewCache::IsCertApplicableToState(const CScCertifi
         return CValidationState::Code::SCID_NOT_FOUND;
     }
 
+    if (!CheckCertTiming(cert.GetScId(), cert.epochNumber))
```

Comment:
add to IsCertApplicableToState an optional bool ptr indicating to the caller if the sender should be banned

Reply:
Done